### PR TITLE
Test e2e for provider in backoffice

### PR DESCRIPTION
**Pull request by Marcelinna**
Add test for adding, editing and deleting provider without resources

Handle exception Failed to open TCP connection from Faraday

### CODE REVIEW

File: app/helpers/url_helper.rb [hunk 1]
```diff
@@ -25,7 +25,7 @@ def self.url_valid?(url)
 
     response = Faraday.get(url)
     response.status == 200
-  rescue URI::InvalidURIError, NoMethodError
+  rescue URI::InvalidURIError, NoMethodError, Faraday::ConnectionFailed
```
Comment (jswk) [inline app/helpers/url_helper.rb:28]: I would actually go a step further and rescue from `Faraday::Error`. Hava look at https://github.com/lostisland/faraday/blob/main/lib/faraday/error.rb, there is for example SSLError, ParsingError, which should also be handled by returning false in that case.
Good catch though!

**Review by jswk — COMMENTED** [review]
I would introduce a single improvement, otherwise it looks good.

File: app/views/backoffice/providers/form/_location.html.haml [hunk 2]
```diff
@@ -3,7 +3,7 @@
     [:street_name_and_number, :postal_code, :city, :region, :country])),
               data: { toggle: "collapse", target: "#location" },
               aria: { expanded: true, controls: "location" } }
-    .card-header.text-left{ id: "marketing-header" }
```
Comment (danielkryska) [inline app/views/backoffice/providers/form/_location.html.haml:6]: Have you checked if the id isn't used anywhere?

File: test/system/cypress/factories/provider.factory.ts [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    contactEmail:string;
+    publicContactsEmail:string;
+    adminFirstName?:string;
+    adminLastName?:string;
+    adminEmail?:string;
+}
+
+export const ProvidersFactory = {
+    create: (args: {[field: string]: string} = {}): IProviders => ({
+        basicName: Utilities.getUUID4(),
```
Comment (danielkryska) [inline test/system/cypress/factories/provider.factory.ts:26]: Good practice would be to remove Utilities related to randomly generated data and use the [faker JS lib](https://github.com/marak/Faker.js/).

File: test/system/cypress/support/providers.ts [hunk 4]
```diff
@@ -0,0 +1,165 @@
+/**
+ * Define new commands types for typescript (for autocompletion)
+ */
+ import { IProviders } from "../factories/provider.factory";
+
+ declare global {
+   namespace Cypress {
+     interface Chainable {
+      
+       fillFormCreateProvider(provider: IProviders, logo:any): Cypress.Chainable<void>;
```
Comment (danielkryska) [inline test/system/cypress/support/providers.ts:10]: 1. provider: Partial<IProvider>
2. Name method as fillForm and use it for creating and updating

**Review by danielkryska — APPROVED** [review]

File: test/system/cypress/support/providers.ts (hunk 4 above)
Comment (Marcelinna) [reply test/system/cypress/support/providers.ts]: I can't change every method for fillForm(), TS does not match them properly, for every test want to use the method for the provider, despite I added correctly another object.

File: app/views/backoffice/providers/form/_location.html.haml (hunk 2 above)
Comment (Marcelinna) [reply app/views/backoffice/providers/form/_location.html.haml:6]: yes, it can be changed to correct name

**Review by jswk — APPROVED** [review]

**Review by kmarszalek — APPROVED** [review]

**Pull request merged**